Assets/World/Global.gd

```diff
@@ -0,0 +1,159 @@
+tool
+extends Node
+class_name Global
+
+enum Factions {
+	NONE,
+	RED,
+	BLUE,
+	GREEN,
+	ORANGE,
+	PURPLE,
+	CYAN,
+	YELLOW,
+	PINK,
+	TEAL,
+	LEMON,
+	BORDEAUX,
+	WHITE,
+	GRAY,
+	BLACK
+}
+
+const FACTIONS = [
+	"None",
+	"Red",
+	"Blue",
+	"Green",
+	"Orange",
+	"Purple",
+	"Cyan",
+	"Yellow",
+	"Pink",
+	"Teal",
+	"Lemon",
+	"Bordeaux",
+	"White",
+	"Gray",
+	"Black"
+]
+
+const COLOR = [
+				Color(0,       0,    0, 0),
+				Color(1,    0.04, 0.04, 1),
+				Color(0,    0.28, 0.71, 1),
+				Color(0,    0.63, 0.09, 1),
+				Color(0.88,  0.4,    0, 1),
+				Color(0.50,    0, 0.50, 1),
+				Color(0,       1,    1, 1),
+				Color(1,    0.84,    0, 1),
+				Color(1,       0,    1, 1),
+				Color(0,    0.57, 0.55, 1),
+				Color(0,       1,    0, 1),
+				Color(0.59, 0.02, 0.16, 1),
+				Color(1,       1,    1, 1),
+				Color(0.50, 0.50, 0.50, 1),
+				Color(0,       0,    0, 1)
+				];
+
+const FACTION_COLOR_NONE =\
+	preload("res://Assets/Player/FactionColor/FactionColorNone.tres")
+const FACTION_COLOR_RED =\
+	preload("res://Assets/Player/FactionColor/FactionColorRed.tres")
+const FACTION_COLOR_BLUE =\
+	preload("res://Assets/Player/FactionColor/FactionColorBlue.tres")
+const FACTION_COLOR_GREEN =\
+	preload("res://Assets/Player/FactionColor/FactionColorGreen.tres")
+const FACTION_COLOR_ORANGE =\
+	preload("res://Assets/Player/FactionColor/FactionColorOrange.tres")
+const FACTION_COLOR_PURPLE =\
+	preload("res://Assets/Player/FactionColor/FactionColorPurple.tres")
+const FACTION_COLOR_CYAN =\
+	preload("res://Assets/Player/FactionColor/FactionColorCyan.tres")
+const FACTION_COLOR_YELLOW =\
+	preload("res://Assets/Player/FactionColor/FactionColorYellow.tres")
+const FACTION_COLOR_PINK =\
+	preload("res://Assets/Player/FactionColor/FactionColorPink.tres")
+const FACTION_COLOR_TEAL =\
+	preload("res://Assets/Player/FactionColor/FactionColorTeal.tres")
+const FACTION_COLOR_LEMON =\
+	preload("res://Assets/Player/FactionColor/FactionColorLemon.tres")
+const FACTION_COLOR_BORDEAUX =\
+	preload("res://Assets/Player/FactionColor/FactionColorBordeaux.tres")
+const FACTION_COLOR_WHITE =\
+	preload("res://Assets/Player/FactionColor/FactionColorWhite.tres")
+const FACTION_COLOR_GRAY =\
+	preload("res://Assets/Player/FactionColor/FactionColorGray.tres")
+const FACTION_COLOR_BLACK =\
+	preload("res://Assets/Player/FactionColor/FactionColorBlack.tres")
+
+const COLOR_MATERIAL = [
+	FACTION_COLOR_NONE,
+	FACTION_COLOR_RED,
+	FACTION_COLOR_BLUE,
+	FACTION_COLOR_GREEN,
+	FACTION_COLOR_ORANGE,
+	FACTION_COLOR_PURPLE,
+	FACTION_COLOR_CYAN,
+	FACTION_COLOR_YELLOW,
+	FACTION_COLOR_PINK,
+	FACTION_COLOR_TEAL,
+	FACTION_COLOR_LEMON,
+	FACTION_COLOR_BORDEAUX,
+	FACTION_COLOR_WHITE,
+	FACTION_COLOR_GRAY,
+	FACTION_COLOR_BLACK
+]
+
+# System variables
+var language = "en"
```

Comment:
Add these lines above `# System Variables`

```
# Godot bug, see https://github.com/godotengine/godot/issues/23207
#warning-ignore-all:unused_class_variable
```

There are many other places where this should be done.